### PR TITLE
v0.2 Segmentation + Detection Upgrade

### PR DESCRIPTION
**Pull request by patrickcleeve2**
- Removed references to Adorned Images and changed to FibsemImage class
- Implemented features to be detected as ABCs:
- Feature(ABC) has a detect method and 4 main attributes: feature_px, feature_m, color_uint8, name
- Updated plotting colours with the option to provide a contrast (plot_det_result_v2 now takes in additional optional argument "inverse" -> defaults to True
- Removed obsolete/deprecated functions in detection.utils

### CODE REVIEW

**Comment by patrickcleeve2** [comment]
@Rohit-Kannachel  please fill in

**Review by LucileNaegele — APPROVED** [review]
Looks good

File: fibsem/segmentation/model.py
Comment (patrickcleeve2) [inline fibsem/segmentation/model.py:1]: change this back to resnet18 as the default

File: fibsem/detection/detection.py [hunk 1]
```diff
@@ -1,45 +1,115 @@
...
+if _DETECTION_V3:
+
+
+    from abc import ABC, abstractmethod
+    # TODO: maybe static methods? probs nicer to use?
+    # e.g. needle_tip = NeedleTip.detect(img, mask, point)
+    # can store color info there too?
+
+    @dataclass
+    class Feature_v3(ABC):
```
Comment (patrickcleeve2) [inline fibsem/detection/detection.py]: rename this from Feature_v3 to Feature as we are just removing the old one,

File: fibsem/detection/utils.py [hunk 2]
```diff
@@ -11,56 +11,46 @@
...
-    FeatureType.LamellaCentre: "red",
-    FeatureType.NeedleTip: "green",
-    FeatureType.LamellaLeftEdge: "magenta",
-    FeatureType.LamellaRightEdge: "orange",
-    FeatureType.LandingPost: "cyan",
-    FeatureType.ImageCentre: "white"
-}
+
+
+# @dataclass
```
Comment (patrickcleeve2) [inline fibsem/detection/utils.py]: remove commented out code

Comment (Rohit-Kannachel) [reply fibsem/detection/utils.py]: removed

File: fibsem/detection/detection.py (hunk 1 above)
Comment (Rohit-Kannachel) [reply fibsem/detection/detection.py]: done

**Pull request merged**